### PR TITLE
Fix binary distribution: add LICENSE and NOTICE

### PR DESCRIPTION
**Pull request by rmetzger**
source should now build without the presence of a .git directory
included the quickstart-archetypes into the main project.

### CODE REVIEW

**Comment by rmetzger** [comment]
I was actually told to push this directly to master, but I'd like to have some feedback on the Job skeleton for the new Java API. 
https://github.com/rmetzger/stratosphere/blob/releasefixes/stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/Job.java

**Comment by StephanEwen** [comment]
I like it. The skeleton for the Java Program is very good in my opinion.

I was thinking that we could even simplify the WordCountExample. The current one is the example one that switches between local data and file inputs. We could have the quickstart example even specialize on one of the two.

@fhueske What do you think?

File: stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/Job.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		
+		
+    	/**
+    	 * Here, you can start creating your execution plan for Stratosphere.
+    	 * 
+    	 * Start with getting some data from the environment, like
+    	 * 	env.readTextFile(textPath);
+    	 * 
+    	 * then, transform the resulting DataSet<String> using operations
```
Comment (fhueske) [inline stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/Job.java]: Maybe a link to the full online documentation would be helpful.

**Comment by fhueske** [comment]
@StephanEwen Jep, I agree. I would go for the CollectionDS version of WordCount such that it can be directly executed. 
I would also add a link to the examples page on the website and point to the examples code in the repository. 

Can we also have such a nice skeleton for the Scala quickstart?

**Comment by rmetzger** [comment]
Okay, I added the links and changed the wordcount.

File: stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import eu.stratosphere.api.java.DataSet;
+import eu.stratosphere.api.java.ExecutionEnvironment;
+import eu.stratosphere.api.java.aggregation.Aggregations;
+import eu.stratosphere.api.java.functions.FlatMapFunction;
+import eu.stratosphere.api.java.tuple.Tuple2;
+import eu.stratosphere.util.Collector;
+
+/**
+ * Implements the "WordCount" program that computes a simple word occurrence histogram
+ * over text files.
```
Comment (fhueske) [inline stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java]: This job does not run on text files.

File: stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import eu.stratosphere.api.java.functions.FlatMapFunction;
+import eu.stratosphere.api.java.tuple.Tuple2;
+import eu.stratosphere.util.Collector;
+
+/**
+ * Implements the "WordCount" program that computes a simple word occurrence histogram
+ * over text files. 
+ * 
+ * <p>
+ * The input is a plain text file with lines separated by newline characters.
```
Comment (fhueske) [inline stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java]: No input ;-)

File: stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	//     PROGRAM
+	// *************************************************************************
+	
+	public static void main(String[] args) throws Exception {
+		
+		// set up the execution environment
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		
+		// get input data
+		DataSet<String> text = getTextDataSet(env);
```
Comment (fhueske) [inline stratosphere-quickstart/quickstart-java/src/main/resources/archetype-resources/src/main/java/WordCountJob.java]: Inline this function?

**Comment by rmetzger** [comment]
Thank you for the feedback @fhueske. I addressed all of your comments.

**Comment by fhueske** [comment]
Looks good to me :+1:

**Comment by rmetzger** [comment]
Ok. WIll merge.

On Tue, May 20, 2014 at 10:15 AM, Fabian Hueske notifications@github.comwrote:

> Looks good to me [image: :+1:]
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/835#issuecomment-43597879
> .

## 

Robert Metzger, Kontakt: metzgerr@web.de, Mobil: 0171/7424461

**Comment by rmetzger** [comment]
Merged in 656022012916b604a3bf955f0a8514ebe3b8d449.

**Pull request closed without merging**